### ISSUE TITLE
Undefined method `unpack1' error with pwshlib 0.5.0

### ISSUE THREAD

**Issue by Phuurl** [puppetlabs/ruby-pwsh#44]
## Describe the Bug
Puppet errors when applying a simple PowerShell command when pwshlib `0.5.0` is installed.
Executing an example with 
```puppet
exec { 'powershell-test':
  command => 'echo hello',
  provider => 'powershell',
}
```
fails with the error
```
Error: undefined method `unpack1' for "H\x00\x00\x00":String
Error: /Stage[main]/Main/Exec[powershell-test]/returns: change from notrun to 0 failed: undefined method `unpack1' for "H\x00\x00\x00":String
```

This behaviour is not experienced if pwshlib is rolled back to `0.4.1`

<img width="972" alt="image" src="https://user-images.githubusercontent.com/1835431/92000014-f3a54b80-ed34-11ea-81ae-4401a6a6e732.png">
<img width="916" alt="image" src="https://user-images.githubusercontent.com/1835431/92000036-f869ff80-ed34-11ea-86da-f2750800079e.png">

## Expected Behavior
The sample PowerShell exec statement is executed, as in `0.4.1`.

## Steps to Reproduce
1. `puppet module install puppetlabs-powershell` - this will pull in pwshlib 0.5.0 automatically
2. `puppet apply` a play with a PowerShell exec statement

## Environment
 - pwshlib 0.5.0, Puppet 4.10.12
 - both Windows 10 and Windows Server 2019

**Comment by Phuurl** [comment]
Also note that after the `undefined method 'unpack1'` error, Puppet will freeze and never return to the command prompt.

**Comment by hdeheer** [comment]
Ran into the same issue. So did some investigating.
`unpack1` is not a function available in ruby 2.1.x (which is the one used by puppet 4.10.x).
`unpack1` got introduced with ruby 2.4. Probably not much you can do besides sticking to 0.4.1 until you upgrade to puppet 5.5.x.

**Comment by michaeltlombardi** [comment]
Piling on, the `puppetlabs-pwshlib` module has support pinned for Puppet versions greater than `5.5.0`:

https://github.com/puppetlabs/ruby-pwsh/blob/main/metadata.json#L79

All versions of Puppet Agent 5.x ship with a ruby version of 2.4 or higher.
